src/types.ts

```diff
@@ -213,6 +213,12 @@ export interface SortIconProps {
   sortDirection: SortDirection | undefined;
 }
 
+export interface SortPriorityProps {
+  priority: number | undefined;
+}
+
+export type SortStatusProps = SortIconProps & SortPriorityProps;
```

Comment:
I believe this is better for 1. performance, 2. catching conflicting types
```suggestion
export interface SortStatusProps extends SortIconProps, SortPriorityProps {};
```